crates/noirc_frontend/src/ast/expression.rs

```diff
@@ -653,3 +653,18 @@ impl Display for FunctionReturnType {
         }
     }
 }
+
+impl FunctionReturnType {
+    pub fn is_equivelent(&self, other: &FunctionReturnType) -> bool {
+        match self {
+            FunctionReturnType::Default(_span) => match other {
+                FunctionReturnType::Default(_span) => true,
+                FunctionReturnType::Ty(_found_type, _span) => false,
+            },
+            FunctionReturnType::Ty(lhs_type, _span) => match other {
+                FunctionReturnType::Default(_span) => false,
```

Comment:
A Default return type should be treated as equivalent to returning a Unit type

Reply:
in that case, I implemented the From rust trait and used it